### PR TITLE
Make unique token unit tests print max reuse on failure only

### PR DESCRIPTION
**Pull request by dalg24**
* Rename {acquire_team_unique_token -> unique_token_team_acquire} for
  consistency

* Move OpenMPTarget macro guards to avoid giving false sense that the
  test ran

Yet another partial fix for #4463

### CODE REVIEW

File: core/unit_test/TestUniqueToken.hpp [hunk 1]
```diff
@@ -268,22 +266,19 @@ class TestAcquireTeamUniqueToken {
...
 
-    ASSERT_EQ(host_errors(0), 0);
+    ASSERT_EQ(host_errors(0), 0) << "max reuse was " << max;
   }
 };
 
-TEST(TEST_CATEGORY, acquire_team_unique_token) {
-  // FIXME_OPENMPTARGET - Not yet implemented.
-#if !defined(KOKKOS_ENABLE_OPENMPTARGET)
+#ifndef KOKKOS_ENABLE_OPENMPTARGET  // FIXME_OPENMPTARGET - Not yet implemented
```
Comment (dalg24) [inline core/unit_test/TestUniqueToken.hpp:278]: Actually I should have disabled only if `TEST_EXECSPACE` is `OpenMPTarget`...

Comment (dalg24) [inline core/unit_test/TestUniqueToken.hpp:278]: Actually I should have disabled only if `TEST_EXECSPACE` is `OpenMPTarget`...

File: core/unit_test/TestUniqueToken.hpp [hunk 2]
```diff
@@ -42,11 +42,11 @@
 //@HEADER
 */
 
-#include <iostream>
+#include <gtest/gtest.h>
 
 #include <Kokkos_Core.hpp>
 
-namespace Test {
+namespace {
```
Comment (masterleinad) [inline core/unit_test/TestUniqueToken.hpp:49]: Why did you change this?

Comment (dalg24) [reply core/unit_test/TestUniqueToken.hpp:49]: Because that's what we should have been doing the whole time

Comment (dalg24) [reply core/unit_test/TestUniqueToken.hpp:49]: This avoids potential collisions at link-time

**Review by masterleinad — APPROVED** [review]

**Comment by dalg24** [comment]
Retest this please

**Review by crtrott — APPROVED** [review]
